posthog/hogql/database/schema/test/test_revenue_analytics.py

```diff
@@ -0,0 +1,273 @@
+                    ("cus_6", Decimal("17476.47254")),
+                ]
+
+    def test_get_revenue_for_schema_source_for_custom_join(self):
+        self.setup_schema_sources()
+        self.team.revenue_analytics_config.persons_join_mode = RevenueAnalyticsPersonsJoinMode.CUSTOM
+        self.team.revenue_analytics_config.persons_join_mode_custom = "id"
+        self.team.revenue_analytics_config.save()
+
+        # These are the 6 IDs inside the CSV files
+        for person_id in [
+            "cus_1_metadata",
+            "cus_2_metadata",
+            "cus_3_metadata",
+            "cus_4_metadata",
+            "cus_5_metadata",
+            "cus_6_metadata",
+        ]:
+            _create_person(team_id=self.team.pk, distinct_ids=[person_id])
+
+        with freeze_time(self.QUERY_TIMESTAMP):
+            queries = [
+                "select revenue_analytics.customer_id, revenue_analytics.revenue from persons order by id asc",
+                "select revenue_analytics.customer_id, $virt_revenue from persons order by id asc",
+            ]
+            for query in queries:
+                response = execute_hogql_query(
+                    parse_select(query),
+                    self.team,
+                    modifiers=self.MODIFIERS,
+                )
+
+                assert response.results == [
+                    ("cus_1", Decimal("429.7424")),
+                    ("cus_2", Decimal("287.4779")),
+                    ("cus_3", Decimal("26182.78099")),
+                    ("cus_4", Decimal("254.12345")),
+                    ("cus_5", Decimal("626.83253")),
+                    ("cus_6", Decimal("17476.47254")),
+                ]
+
+    def test_get_revenue_for_schema_source_for_customer_with_multiple_distinct_ids(self):
+        self.setup_schema_sources()
+        self.team.revenue_analytics_config.persons_join_mode = RevenueAnalyticsPersonsJoinMode.EMAIL
+        self.team.revenue_analytics_config.save()
+
+        # Person has several distinct IDs, but only one of them can be matched from the customer table
+        _create_person(team_id=self.team.pk, distinct_ids=["distinct_1", "john.doe@example.com"])
+        breakpoint()
+
+        with freeze_time(self.QUERY_TIMESTAMP):
+            response = execute_hogql_query(
+                parse_select("select *, $virt_revenue from persons"),
+                self.team,
+                modifiers=self.MODIFIERS,
+            )
+
+            # TODO: This is failing for more than one reason:
+            # 1. I can only get the revenue to work if I select * first, doesnt work if select just revenue
+            # 2. This is returning an entry for each of the distinct IDs which is understandable but could probably get the one that exists only
+            # 3. If I change the select to a specific person `where id = {}` then it fails to return for the matching email, never returns revenue
+            assert response.results[0][6] == Decimal("429.7424")
```

Comment:
@mariusandra This is the test where I can reproduce the problem, it only happens when I have a person with more than one distinct_id. The easiest way to debug this is by adding a breakpoint here and running queries via `execute_hogql_query`. Make sure you include `modifiers=self.MODIFIERS` to be able to parse the mocked data warehouse tables appropriately

Reply:
I'm unable to run the tests in my debugger in pycharm for what looks like unrelated reasons.

```
AttributeError: module 'django.conf.global_settings' has no attribute 'CLICKHOUSE_CONN_POOL_MAX'
```

Can you tell me what query does this generate? Let's start by making sure it tries to do the right thing before debugging. So... what does it generate, and what should it generate?